### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by gonziesc**
## Summary
- Implementing authentication
- Creating sessions endpoint for users

## Trello Card
https://trello.com/c/2oMQC2R4/26-implementar-autenticaci%C3%B3n

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -34,6 +34,10 @@ gem 'jbuilder', '~> 2.5'
 # gem 'capistrano-rails', group: :development
 gem 'devise', '~> 4.0'
 
+gem 'versionist'
```
Comment (hdf1986) [inline Gemfile]: Can version this gem?

File: Gemfile [hunk 2]
```diff
@@ -34,6 +34,10 @@ gem 'jbuilder', '~> 2.5'
...
+
+gem 'jwt'
```
Comment (hdf1986) [inline Gemfile]: Same here

File: app/authentication/authentication_token_manager.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# app/authentication/authentication_token_manager.rb
+class AuthenticationTokenManager
+  class << self
+    def encode(payload)
+      JWT.encode(payload, Rails.application.secrets.secret_key_base)
```
Comment (hdf1986) [inline app/authentication/authentication_token_manager.rb]: I think you need to explicitly set your encoding method here, did you tried the endpoints? As far as i remember this should break when executed if it has no encryption method

Comment (gonziesc) [reply app/authentication/authentication_token_manager.rb]: I tried `api/v1/users/sessions` with a valid user password combination and it successfully returned the access token and the renew_id. Also tested with a non valid user and show the error correctly. Is there anything else I need to try?

**Pull request merged**